### PR TITLE
Bug 1261169 - Store remote clients version

### PR DESCRIPTION
**Pull request by eoger**

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
BrowserTable.swift | 690 | Force casts should be avoided. 
RemoteTabsTable.swift | 66 | Force casts should be avoided. 
RemoteTabsTable.swift | 67 | Force casts should be avoided. 
RemoteTabsTable.swift | 148 | Force casts should be avoided. 
RemoteTabsTable.swift | 149 | Force casts should be avoided. 
RemoteTabsTable.swift | 99 | Force tries should be avoided. 
RemoteTabsTable.swift | 105 | Force tries should be avoided. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Storage/SQL/BrowserTable.swift [hunk 1]
```diff
@@ -917,6 +917,14 @@ open class BrowserTable: Table {
             }
         }
 
+        if from < 25 && to >= 25 {
+            if !self.run(db, queries: [
+                "ALTER TABLE \(TableClients) ADD COLUMN version TEXT"
```
Comment (rnewman) [inline Storage/SQL/BrowserTable.swift]: This should be moved to `RemoteClientsTable`. Put it in an overriding implementation of `updateTable`. Leave the `BrowserTable` version at 24.

File: Storage/Clients.swift [hunk 2]
```diff
@@ -37,18 +37,18 @@ public struct RemoteClient: Equatable {
         self.device = json["device"].stringValue
```
Comment (rnewman) [inline Storage/Clients.swift]: Ugh, all of these supposedly optional `.stringValue` lines — everything but name — should be `.string`. Fucking SwiftyJSON. @sleroux, do you agree?

These used to be `asString`, which was an optional.

Comment (sleroux) [inline Storage/Clients.swift]: Yup I think this is supposed to be `.string`. We have so many stringifying methods!

**Comment by eoger** [comment]
Thanks for the feedback, commit amended.

File: Storage/SQL/RemoteTabsTable.swift [hunk 3]
```diff
@@ -8,9 +8,20 @@ import Shared
...
 let TableTabs = "tabs"
 
+private let log = Logger.syncLogger
+
 class RemoteClientsTable<T>: GenericTable<RemoteClient> {
     override var name: String { return TableClients }
-    override var version: Int { return 1 }
+    override var version: Int { return 2 }
+
+    func run(_ db: SQLiteDBConnection, sql: String, args: Args? = nil) -> Bool {
```
Comment (rnewman) [inline Storage/SQL/RemoteTabsTable.swift]: You only use this once; just inline and simplify this logic at point of use.

File: Storage/SQL/RemoteTabsTable.swift [hunk 4]
```diff
@@ -65,13 +70,34 @@ class RemoteClientsTable<T>: GenericTable<RemoteClient> {
...
+            log.debug("Skipping update from \(from) to \(to).")
+            return true
+        }
+
+        if from < 2 && to >= 2 {
+            let sql = "ALTER TABLE \(TableClients) ADD COLUMN version TEXT"
+            let err = db.executeChange(sql)
+            if err != nil {
+                log.error("Error running SQL in RemoteClientsTable: \(err?.localizedDescription ?? "nil")")
+                log.error("SQL was \(sql)")
```
Comment (rnewman) [inline Storage/SQL/RemoteTabsTable.swift]: This line isn't necessary. Kill this and land it!

**Comment by eoger** [comment]
I don't really know how to react regarding that failing build, I don't think my changes triggered it... Anyway this is ready for merge.

**Comment by rnewman** [comment]
@sleroux, would you?

**Comment by sleroux** [comment]
Looks like the latest failure is from an orange test we have so I would ignore that.

The failure before that looks to be codesigning related. Could you try rebasing your branch from the latest master? 

It doesn't look like it's failing because of anything you did though so I wouldn't worry about that.

**Comment by eoger** [comment]
It is already at the tip actually

**Comment by sleroux** [comment]
Looks like the same issue is happening on master. Investigating...

**Comment by eoger** [comment]
Closing because buddybuild is dumb

**Pull request closed without merging**